models/migrations/v114.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"net/url"
+	"strings"
+
+	"xorm.io/xorm"
+)
+
+func sanitizeOriginalURL(x *xorm.Engine) error {
+
+	type Repository struct {
+		ID          int64
+		OriginalURL string `xorm:"VARCHAR(2048)"`
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	var last int
+	const batchSize = 50
+	for {
+		var results = make([]Repository, 0, batchSize)
+		err := x.Where("original_url <> '' AND original_url IS NOT NULL").
+			And("original_service_type = 0 OR original_service_type IS NULL").
+			OrderBy("id").
+			Limit(batchSize, last).
+			Find(&results)
+		if err != nil {
+			return err
+		}
+		if len(results) == 0 {
+			break
+		}
+		last += len(results)
+
+		for _, res := range results {
+			u, err := url.Parse(res.OriginalURL)
+			if err != nil {
+				// it is ok to continue here, we only care about fixing URLs that we can read
+				continue
+			}
+
+			if len(u.User.Username()) > 0 {
+				pass, _ := u.User.Password()
+				userAuth := u.User.Username() + ":" + pass + "@"
+				OriginalURL := strings.Replace(res.OriginalURL, userAuth, "", -1)
```

Comment:
How about set `u.User = nil` and `originalURL = u.String()`?

Reply:
Same as above for u.User = nil and u.String() still has the auth info inside of it so we can't use that

Reply:
I changed the code a bit to use values from u to rebuild URL instead of using strings.